packages/ui/src/components/shared/__tests__/SettingsWalletButton.spec.tsx

```diff
@@ -0,0 +1,71 @@
+import { shallowEqual } from 'react-redux';
+import { initializeKeyring, newUser, render, screen, UserEvent, waitFor } from '__tests__/testUtils';
+import SettingsWalletButton from '../SettingsWalletButton';
+
+describe('SettingsWalletButton', () => {
+  it('should hide the dialog by default', () => {
+    render(<SettingsWalletButton />, { preloadedState: { app: { seedReady: true, ready: true, locked: false } } });
+    expect(screen.queryByText('Settings')).not.toBeInTheDocument();
```

Comment:
We should check for role `dialog` to make sure it is a dialog not just any kind of text that has `Settings`:
```suggestion
    expect(await screen.findByRole('dialog', { name: /Settings/ })).not.toBeInTheDocument();
```

Same for other places as well.